.github/workflows/benchmarks.yml

```diff
@@ -5,8 +5,30 @@ permissions: {}
 jobs:
   telio-firewall:
       runs-on: ubuntu-22.04
+      env:
+        CARGO_TARGET_DIR: "/home/runner/work/libtelio/libtelio/target/criterion"
       steps:
         - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+        - name: Install critcmp for benchmark comparison
+          run: cargo install critcmp
         - name: Telio firewall benchmarks
           working-directory: crates/telio-firewall
-          run: cargo bench --features test_utils --bench firewall_bench "64" -- --warm-up-time 1 --measurement-time 1
+          run: cargo bench --features test_utils --bench firewall_bench "64" -- --warm-up-time 1 --measurement-time 1 --save-baseline current
```

Comment:
I think that this (and the other one) benchmark run should be run on the same physical core (and only on that one) to avoid noise from scheduling (can be done with `taskset`), but this only makes sense when we run on a known machine where we can prevent other processes from being scheduled on our selected core.

Overall, I think this is a good source of ideas for running benchmarks in CI: https://manuel.bernhardt.io/posts/2023-11-16-core-pinning/